src/main/java/com/jcabi/github/Status.java

```diff
@@ -52,57 +53,82 @@
      * Get URL.
      * @return URL as string.
      */
-    String targetUrl();
+    Optional<String> targetUrl();
 
     /**
      * Get description.
      * @return Description as string.
      */
-    String description();
+    Optional<String> description();
 
     /**
      * Get context.
      * @return Context as string
      */
-    String context();
+    Optional<String> context();
 
     /**
      * States of Status API.
      * @author Marcin Cylke(marcin.cylke+github@gmail.com)
      * @version $Id$
      */
-    public enum State {
+    enum State implements StringEnum {
         /**
          * Pending state.
          */
-        Pending,
+        Pending("pending"),
         /**
          * Success state.
          */
-        Success,
+        Success("success"),
         /**
          * Error state.
          */
-        Error,
+        Error("error"),
         /**
          * Failure state.
          */
-        Failure;
+        Failure("failure");
 
         /**
-         * Get enum value from string.
-         * @param name Enum name
-         * @return Matched enum name
+         * Commit status state.
          */
-        public static State forValue(final String name) {
+        private final transient String state;
+
+        /**
+         * Ctor.
+         * @param stat Commit status state string
+         */
+        State(final String stat) {
+            this.state = stat;
+        }
+
+        /**
+         * Get enum value from identifier string.
+         * @param ident Commit status state string
+         * @return Corresponding State
+         */
+        public static State forValue(
```

Comment:
@caarlos0 Again, I prefer to be explicit and avoid this sort of magic and dependency on Locale (for case conversion).